### PR TITLE
Added supporting for execution recipes locally using maven plugin

### PR DESCRIPTION
**Pull request by prakashmrec**
Also, added more configuration parameters to be able to execute some tests locally and some using TestServer. See the updated pom of maven-plugin-tester module.

### CODE REVIEW

File: modules/maven-plugin-tester/pom.xml [hunk 1]
```diff
@@ -25,14 +25,24 @@
...
-                    <username>demoUser</username>
-                    <password>demoPassword</password>
-                    <server>http://testserver.readyapi.io:8080</server>
-                </configuration>
                 <executions>
                     <execution>
-                        <id>run</id>
+                        <id>RunLocally</id>
+                        <configuration>
+                            <runXmlProjects>false</runXmlProjects>
```
Comment (olensmar) [inline modules/maven-plugin-tester/pom.xml]: is this required? Couldn't we automatically set it to false internally for local executions? Also - what happens if you set it to true for local?

Comment (prakashmrec) [inline modules/maven-plugin-tester/pom.xml]: Yes, we could skip the project execution with a warning message in the logs but then user might think that his tests are working. So, I thought it will be better to fail the build if there are projects to execute with local execution, that will give more clear feedback to the user.

Comment (prakashmrec) [inline modules/maven-plugin-tester/pom.xml]: The build will fail if it is set to true with local execution.

Comment (olensmar) [inline modules/maven-plugin-tester/pom.xml]: ok - i was thinking that the user shouldn't have to configure anything for local execution - if they specify projects to execute the build should simply fail as you indicate - and if they add testserver credentials the plugin will automatically use remote execution

Comment (prakashmrec) [inline modules/maven-plugin-tester/pom.xml]: Ok, so should we remove the default value then for projects directory? or should we check if there is any project in default projects directory then fail for local execution?

Comment (prakashmrec) [inline modules/maven-plugin-tester/pom.xml]: OR just ignore projects from default projects directory.

File: modules/maven-plugin-tester/pom.xml [hunk 2]
```diff
@@ -29,7 +29,7 @@
                     <execution>
                         <id>RunLocally</id>
                         <configuration>
-                            <runXmlProjects>false</runXmlProjects>
+                            <ignoreXmlProjects>true</ignoreXmlProjects>
```
Comment (olensmar) [inline modules/maven-plugin-tester/pom.xml]: maybe we should call this ignoreProjectFiles instead - using "Xml" here might be confusing!?

File: modules/groovy-dsl/src/test/groovy/com/smartbear/readyapi4j/dsl/execution/RecipeExecutorAdaptor.groovy [hunk 3]
```diff
@@ -44,4 +45,9 @@ class RecipeExecutorAdaptor implements RecipeExecutor {
     void removeRecipeFilter(RecipeFilter recipeFilter) {
 
     }
+
+    @Override
+    ExecutionMode getExecutionMode() {
+        return null
```
Comment (spindelmanne) [inline modules/groovy-dsl/src/test/groovy/com/smartbear/readyapi4j/dsl/execution/RecipeExecutorAdaptor.groovy]: Should we use Optional<ExecutionMode>? Or do we maybe need to be 1.7 compatible?

File: modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java [hunk 4]
```diff
@@ -133,29 +157,50 @@ public void execute()
...
-                    recipeCount++;
-                    response = runJsonRecipe(f);
-                } else if (fileName.endsWith(".xml")) {
-                    projectCount++;
-                    response = runXmlProject(f);
-                } else {
-                    getLog().warn("Unexpected filename: " + fileName);
-                    continue;
-                }
+            //Run recipes
```
Comment (spindelmanne) [inline modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java]: Extract method instead of inline comment?

File: modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java [hunk 5]
```diff
@@ -133,29 +157,50 @@ public void execute()
...
-                } catch (MojoFailureException exception) {
-                    failCount++;
+                    try {
+                        handleResponse(response, report, file);
+                    } catch (MojoFailureException exception) {
+                        failCount++;
+                    }
                 }
             }
+            //Run XML projects
```
Comment (spindelmanne) [inline modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java]: Extract method instead of inline comment?

File: modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java [hunk 6]
```diff
@@ -133,29 +157,50 @@ public void execute()
...
+            if (shouldRunProjects() && xmlProjectFiles != null) {
+                for (String file : xmlProjectFiles) {
+                    String fileName = file.toLowerCase();
+
+                    File f = new File(xmlProjectDirectory, file);
```
Comment (spindelmanne) [inline modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java]: Very non-descriptive variable name

File: modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java [hunk 7]
```diff
@@ -186,6 +231,18 @@ public void execute()
...
 
+    private boolean shouldRunProjects() {
+        return !ignoreProjectFiles;
+    }
+
+    private boolean shouldRunRecipes() {
+        return !ignoreRecipes;
+    }
+
+    private boolean notPresent(List<String> recipeFiles) {
```
Comment (spindelmanne) [inline modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java]: It isn't always recipe files we're using this for

File: modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java [hunk 8]
```diff
@@ -186,6 +231,18 @@ public void execute()
         }
     }
 
+    private boolean shouldRunProjects() {
```
Comment (olensmar) [inline modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java]: not a merge-breaker - but these two methods seem somewhat redundant!? :-)

Comment (spindelmanne) [inline modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java]: I disagree. It's nice to attach a name to a condition.

Comment (prakashmrec) [inline modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java]: I added these for better readability :)

Comment (olensmar) [inline modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java]: ok - then ignore my comment :-)

File: modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java (hunk 4 above)
Comment (prakashmrec) [inline modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java]: I tried to extract method, but that will require much more refactoring. So, I gave up, probably we should do it in a separate PR.

File: modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java (hunk 7 above)
Comment (prakashmrec) [inline modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java]: true, will fix.

File: modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java (hunk 6 above)
Comment (prakashmrec) [inline modules/maven-plugin/src/main/java/com/smartbear/readyapi4j/maven/RunMojo.java]: :)

File: modules/groovy-dsl/src/test/groovy/com/smartbear/readyapi4j/dsl/execution/RecipeExecutorAdaptor.groovy (hunk 3 above)
Comment (prakashmrec) [inline modules/groovy-dsl/src/test/groovy/com/smartbear/readyapi4j/dsl/execution/RecipeExecutorAdaptor.groovy]: We can use optional. Source level is set to 1.8 in the root pom.

Comment (prakashmrec) [inline modules/groovy-dsl/src/test/groovy/com/smartbear/readyapi4j/dsl/execution/RecipeExecutorAdaptor.groovy]: Actually it should not be optional, I will fix it by returning some value.
